grammar.js

```diff
@@ -1688,15 +1688,20 @@ module.exports = grammar({
     ),
     pragma_directive: $ => seq('pragma',
         choice(
-            seq('warning', choice('disable', 'restore'), commaSep($.preproc_warning_number)),
+            seq('warning',
+              choice('disable', 'restore'),
+              commaSep(
+                  choice(
+                    $.identifier,
+                    alias($.preproc_integer_literal, $.integer_literal),
```

Comment:
This was `preproc_warning_number`, but you changed this to `preproc_integer_literal`. Was this intentional? Would it be possible to use warning identifiers such as "CS0219" here?

Reply:
It allows identifier as well. I thought between identifier and integer we'd have good coverage but I should probably just put a regex in here.

Reply:
Hmm, I actually quite prefer the message numbers showing up as either an integer_literal or an identifier depending on whether they are purely numeric or start with letters so will stick with this unless there are objections.